### ISSUE TITLE
service, port, protocol, etc should accept NOT

### ISSUE THREAD

**Issue by luizluca** [firewalld/firewalld#843]
**What would you like to be added**:

Only source and destination support NOT. However, they are valid situations where NOT could be used  with other filters like service, protocol, etc.

And that kind of negative match is supported by backends (iptables, nftables)

**Why is this needed**:

I would like a way to write with rich language a rule like:

```
ip saddr @my_scanners meta l4proto != icmp ct state { new, untracked } drop
```

Related: #842

**Comment by erig0** [comment]
You can mimic the `not` behavior with two rich rules.

Instead of using
```
rule priority=32767 source ipset="my_scanners" not protocol value=icmp drop
```
Use this:

```
rule priority=32767 source ipset="my_scanners" protocol value=icmp accept
rule priority=32766 source ipset="my_scanners" drop
```

**Comment by luizluca** [comment]
> You can mimic the `not` behavior with two rich rules.
> 
> Instead of using
> 
> ```
> rule priority=32767 source ipset="my_scanners" not protocol value=icmp drop
> ```
> 
> Use this:
> 
> ```
> rule priority=32767 source ipset="my_scanners" protocol value=icmp accept
> rule priority=32766 source ipset="my_scanners" drop
> ```

Yes, and I'm already doing it. However, it is assuming I already know the result of ICMP (or whatever I want to exclude from the rule) will have further in the ruleset. I would expect a solution that didn't make assumptions on what will come after. And don't think exclusively about this ICMP case. A NOT condition is useful for many cases.
